plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/DataMapperMediatorItemProvider.java

```diff
@@ -308,42 +294,45 @@ public Object getImage(Object object) {
-        return label == null || label.length() == 0 ?
-            getString("_UI_DataMapperMediator_type") :
-            getString("_UI_DataMapperMediator_type") + " " + label;
+        String label = ((DataMapperMediator) object).getDescription();
+        return label == null || label.length() == 0 ? getString("_UI_DataMapperMediator_type")
+                : getString("_UI_DataMapperMediator_type") + " " + label;
     }
 
     /**
      * This handles model notifications by calling {@link #updateChildren} to update any cached
      * children and by creating a viewer notification, which it passes to {@link #fireNotifyChanged}.
      * <!-- begin-user-doc -->
      * <!-- end-user-doc -->
+     * 
      * @generated
      */
     @Override
     public void notifyChanged(Notification notification) {
         updateChildren(notification);
 
         switch (notification.getFeatureID(DataMapperMediator.class)) {
-            case EsbPackage.DATA_MAPPER_MEDIATOR__INPUT_TYPE:
-            case EsbPackage.DATA_MAPPER_MEDIATOR__CONFIGURATION_LOCAL_PATH:
-            case EsbPackage.DATA_MAPPER_MEDIATOR__INPUT_SCHEMA_LOCAL_PATH:
-            case EsbPackage.DATA_MAPPER_MEDIATOR__OUTPUT_SCHEMA_LOCAL_PATH:
-            case EsbPackage.DATA_MAPPER_MEDIATOR__OUTPUT_TYPE:
-                fireNotifyChanged(new ViewerNotification(notification, notification.getNotifier(), false, true));
-                return;
-            case EsbPackage.DATA_MAPPER_MEDIATOR__INPUT_CONNECTOR:
-            case EsbPackage.DATA_MAPPER_MEDIATOR__OUTPUT_CONNECTOR:
-            case EsbPackage.DATA_MAPPER_MEDIATOR__CONFIGURATION:
-            case EsbPackage.DATA_MAPPER_MEDIATOR__INPUT_SCHEMA:
-            case EsbPackage.DATA_MAPPER_MEDIATOR__OUTPUT_SCHEMA:
-                fireNotifyChanged(new ViewerNotification(notification, notification.getNotifier(), true, false));
-                return;
+        case EsbPackage.DATA_MAPPER_MEDIATOR__INPUT_TYPE:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__CONFIGURATION_LOCAL_PATH:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__INPUT_SCHEMA_LOCAL_PATH:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__OUTPUT_SCHEMA_LOCAL_PATH:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__XSLT_STYLE_SHEET_LOCAL_PATH:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__OUTPUT_TYPE:
+            fireNotifyChanged(new ViewerNotification(notification, notification.getNotifier(), false, true));
+            return;
+        case EsbPackage.DATA_MAPPER_MEDIATOR__INPUT_CONNECTOR:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__OUTPUT_CONNECTOR:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__CONFIGURATION:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__INPUT_SCHEMA:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__OUTPUT_SCHEMA:
+        case EsbPackage.DATA_MAPPER_MEDIATOR__XSLT_STYLE_SHEET:
+            fireNotifyChanged(new ViewerNotification(notification, notification.getNotifier(), true, false));
```

Comment:
Are we intentionally going to change the notification firing from case EsbPackage.DATA_MAPPER_MEDIATOR__OUTPUT_SCHEMA to case EsbPackage.DATA_MAPPER_MEDIATOR__XSLT_STYLE_SHEET ?

Reply:
This is a generated code block.